src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double INCREASE_COEFICIENT = 1.5;
+    private T[] array;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        capacity = 10;
+        array = (T[]) new Object[capacity];
+    }
 
     @Override
     public void add(T value) {
-
+        if (!isEnoughSpace()) {
+            relocateArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of range");
+        } else if (index == size) {
```

Comment:
```suggestion
        }
         if (index == size) {
```
you don't need else after exception or return in if